### ISSUE TITLE
React to breaking NuGet changes

### ISSUE THREAD

**Issue by natemcmaster** [aspnet/BuildTools#544]
Some of our MSBuild tasks are failing on the newest .NET Core SDK due to breaking changes in NuGet 4.6.0.

cref https://github.com/NuGet/Home/issues/6398

Example:
```
C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\modules\KoreBuild.Tasks\module.targets(105,5): 
error MSB4018: System.MissingMethodException: Method not found: 'System.Threading.Tasks.Task`1<Boolean> NuGet.Protocol.Core.Types.MetadataResource.Exists(System.String, NuGet.Common.ILogger, System.Threading.CancellationToken)'. [C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\KoreBuild.proj]
   at KoreBuild.Tasks.UpgradeDependencies.<GetPackageVersion>d__20.MoveNext() [C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\KoreBuild.proj]
   at System.Runtime.CompilerServices.AsyncTaskMethodBuilder`1.Start[TStateMachine](TStateMachine& stateMachine) [C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\KoreBuild.proj]
   at KoreBuild.Tasks.UpgradeDependencies.GetPackageVersion(VersionRange range) [C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\KoreBuild.proj]
   at KoreBuild.Tasks.UpgradeDependencies.<ExecuteAsync>d__19.MoveNext() in C:\b\w\9571b3e5f03d051f\modules\KoreBuild.Tasks\UpgradeDependencies.cs:line 90 [C:\b\w\e86c75abf8916d0d\.dotnet\buildtools\korebuild\2.1.0-preview1-15661\KoreBuild.proj]
```

**Comment by natemcmaster** [comment]
@ryanbrandenburg updating this in deps.props should upgrade our tools to match the sdk

```
<Tooling_NuGetBuildTasksPackageVersion>4.6.0-rtm-4789</Tooling_NuGetBuildTasksPackageVersion>
```

**Comment by ryanbrandenburg** [comment]
Is there a concern about allowing back-compat with older tooling, or can we just move forward?

**Comment by natemcmaster** [comment]
I think we're okay to move forward. The only code path affected AFAIK is command-line. If you find any task assemblies or tools that get loaded into VS, then we may have an issue.

**Comment by natemcmaster** [comment]
To clarify things, we have two options (assuming @nuget isn't going to fix these changes).

1. Update our task assemblies. We get loaded in the same MSBuild process as NuGet, so we have to try and match their assembly verison.
2. Move our task assemblies into a separate process or load context. Tools like NuGetPackageVerifier run out-of-proc from MSBuild and are isolated from NuGet API breaks. It might be a more sane model for build tools going forward.

**Comment by ryanbrandenburg** [comment]
I've done 1 for now in #546 because I'd already gotten most of the way done and want to get people unblocked. If we feel strongly about 2 or this bites us again I can do 2 in a separate PR.

**Comment by Eilon** [comment]
Taking a gamble and closing this because @ryanbrandenburg did the workaround. Should we hit this again, we'll do option (2). Current guesstimate is ~2 weeks of work for that.
